src/enumerable.cr

```diff
@@ -198,6 +198,34 @@ module Enumerable(T)
     n.times { each { |x| yield x } }
   end
 
+  # Traverses the depth of a structure and returns the value.
+  # Returns nil if not found.
+  #
+  # ```
+  # h = {"a" => {"b" => [10, 20, 30]}}
+  # h.dig "a", "b", 1             # => 20
+  # h.dig "a", "b", "c", "d", "e" # => nil
+  # ```
+  def dig(key : K, *subkeys)
+    if (value = self[key]?) && value.responds_to?(:dig)
+      value.dig(*subkeys)
+    end
+  end
+
+  def dig(key : K)
+    self[key]?
+  end
+
+  def dig(key : Int32, *subkeys)
+    if (value = self[key]?) && value.responds_to?(:dig)
+      value.dig(*subkeys)
+    end
+  end
+
+  def dig(key : Int32)
```

Comment:
My bad, sorry! There might second definition needed (`dig(index : Int)`) due to `[]?(index : Int)` defined in `Indexable(T)` module, which does not take `T` as argument type.